symphony-bdk-http/symphony-bdk-http-webclient/src/main/java/com/symphony/bdk/http/webclient/ApiClientWebClient.java

```diff
@@ -146,7 +147,9 @@ public ApiClientWebClient(final WebClient webClient, String basePath, Map<String
       if (unwrap instanceof ApiException) {
         throw (ApiException) unwrap;
       } if(e.getCause() instanceof ConnectTimeoutException){
-        throw new RuntimeException(new SocketTimeoutException(e.getMessage()));
+        WebClientRequestException exception = (WebClientRequestException) e;
+        throw new WebClientRequestException(new SocketTimeoutException(e.getMessage()), exception.getMethod(),
```

Comment:
should it end up in a ConnectException?

Reply:
No actually the ConnectTimeoutException corresponds to a "timeout" therefore SocketTimeoutException. If "connection refused" it will throw WebClientRequestException enclosing a ConnectException which is already good.